src/coreComponents/physicsSolvers/simplePDE/LaplaceVEM.hpp

```diff
@@ -0,0 +1,178 @@
+                             DomainPartition & domain ) override;
+
+  virtual void
+  implicitStepSetup( real64 const & time_n,
+                     real64 const & dt,
+                     DomainPartition & domain ) override;
+
+  virtual void
+  setupDofs( DomainPartition const & domain,
+             DofManager & dofManager ) const override;
+
+  // virtual void
+  // setupSystem( DomainPartition & domain,
+  //              DofManager & dofManager,
+  //              CRSMatrix< real64, globalIndex > & localMatrix,
+  //              array1d< real64 > & localRhs,
+  //              array1d< real64 > & localSolution,
+  //              bool const setSparsity = false ) override;
+
+  virtual void
+  assembleSystem( real64 const time,
+                  real64 const dt,
+                  DomainPartition & domain,
+                  DofManager const & dofManager,
+                  CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                  arrayView1d< real64 > const & localRhs ) override;
+
+  virtual void
+  applyBoundaryConditions( real64 const time,
+                           real64 const dt,
+                           DomainPartition & domain,
+                           DofManager const & dofManager,
+                           CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                           arrayView1d< real64 > const & localRhs ) override;
+
+  virtual void
+  solveSystem( DofManager const & dofManager,
+               ParallelMatrix & matrix,
+               ParallelVector & rhs,
+               ParallelVector & solution ) override;
+
+  virtual void
+  applySystemSolution( DofManager const & dofManager,
+                       arrayView1d< real64 const > const & localSolution,
+                       real64 const scalingFactor,
+                       DomainPartition & domain ) override;
+
+  virtual void
+    resetStateToBeginningOfStep( DomainPartition & GEOSX_UNUSED_PARAM( domain ) ) override;
+
+  virtual void
+  implicitStepComplete( real64 const & time,
+                        real64 const & dt,
+                        DomainPartition & domain ) override;
+
+  //END_SPHINX_INCLUDE_03
+  /**@}*/
+
+  // This method is specific to this Laplace solver
+  // It is used to apply Dirichlet boundary condition
+  // and called when the base class applyBoundaryConditions() is called
+  void applyDirichletBCImplicit( real64 const time,
+                                 DofManager const & dofManager,
+                                 DomainPartition & domain,
+                                 CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                                 arrayView1d< real64 > const & localRhs );
+
+  // Choice of transient treatment options (steady, backward, forward Euler scheme):
+  //START_SPHINX_INCLUDE_01
+  enum class TimeIntegrationOption : integer
+  {
+    SteadyState,
+    ImplicitTransient,
+    ExplicitTransient
```

Comment:
I don't know why we have the explicit option for Laplace at all.  Perhaps something to remove when we put the base class in place.